library/query_self_hosted_workflow.yml

```diff
@@ -0,0 +1,25 @@
+id: RP-10
+
+info:
+  name: Self Hosted Run
+  severity: high
+  description: Self Hosted runners should not be used for public repositories.
+  full-description: |
+    Self Hosted runners do not have the same security controls as GitHub Hosted runners and
+    do not have the guarantees of a clean ephemeral environment. Self Hosted runners should
+    only be used for private repositories.
+  references:
+    - https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#hardening-for-self-hosted-runners
+  tags:
+    - unauthenticated
+
+query: |
+  MATCH (w:Workflow)-[*]->(j:Job)
+    WHERE (
+        "issues" in w.trigger OR
+        "issue_comment" in w.trigger OR
+        "pull_request_target" in w.trigger
+    ) AND (
+        j.machine = "self-hosted"
+    )
+  RETURN DISTINCT w.url AS url;
```

Comment:
I would remove the trigger limitation
The permission limitation for hosted runners aren't applied for self-hosted ones.
so even for `pull_request` the job may have high permissions